dwarf.c

```diff
@@ -1787,15 +1795,16 @@ resolve_unit_addrs_overlap (struct backtrace_state *state,
 				error_callback, data, &new_vec.vec);
   if (grew == NULL)
     return 0;
+  memset (&enclosing, 0, sizeof enclosing);
 
-  from = 0;
-  to = 0;
-  resolve_unit_addrs_overlap_walk (state, &from, &to, NULL, addrs_vec,
+  walk_ok = resolve_unit_addrs_overlap_walk (state, &enclosing, addrs_vec,
 				   error_callback, data, &new_vec);
   backtrace_vector_free (state, &addrs_vec->vec, error_callback, data);
-  *addrs_vec = new_vec;
+  backtrace_vector_free (state, &enclosing, error_callback, data);
+  if (walk_ok)
+    *addrs_vec = new_vec;
```

Comment:
Previously return code from `walk` subroutine was ignored here. Was that intended?